### PR TITLE
Bugfix/66 gracefull crash handling

### PR DESCRIPTION
**Pull request by AssemblyJohn**
## Describe your changes

Updated the following:
- generate_csr_interface for clean handling
- handled some various unhandled errors
- updated return types
- removed OpenSSL warnings
- removed unnecessary test

## Issue ticket number and link

https://github.com/EVerest/libevse-security/issues/66

Supersedes: https://github.com/EVerest/libevse-security/pull/64 as it includes all changes placed there.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [ ] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: lib/evse_security/evse_security.cpp [hunk 1]
```diff
@@ -1168,21 +1408,25 @@ int EvseSecurity::get_leaf_expiry_days_count(LeafCertificateType certificate_typ
...
+    if (key_pair.status == GetCertificateInfoStatus::Accepted) {
         try {
             fs::path certificate_path;
 
-            if (key_pair.pair.value().certificate.empty() == false)
-                certificate_path = key_pair.pair.value().certificate;
-            else
-                certificate_path = key_pair.pair.value().certificate_single;
+            if (key_pair.info.has_value()) {
+                if (key_pair.info.value().certificate.has_value())
```
Comment (Pietfried) [inline lib/evse_security/evse_security.cpp:1417]: please use `{` and `}`

File: tests/openssl_supplier_test.cpp [hunk 2]
```diff
@@ -41,15 +41,6 @@ TEST_F(OpenSSLSupplierTest, generate_key_RSA_3072) {
     ASSERT_TRUE(res);
 }
 
-TEST_F(OpenSSLSupplierTest, generate_key_RSA_7680) {
```
Comment (Pietfried) [inline tests/openssl_supplier_test.cpp:44]: Why was this removed?

File: tests/tests.cpp [hunk 3]
```diff
@@ -21,7 +21,7 @@
 
 #if USING_OPENSSL_3
 // provider management has changed - ensure tests still work
-#ifndef USING_TPM2
+#ifndef USING_TPM2contains_certificate_hash
```
Comment (Pietfried) [inline tests/tests.cpp:24]: I guess this is not intended?

**Review by Pietfried — COMMENTED** [review]
lgtm, see some comments inline

File: tests/openssl_supplier_test.cpp (hunk 2 above)
Comment (AssemblyJohn) [reply tests/openssl_supplier_test.cpp:44]: It took almost 1 minute for generation. Keys of that size are supposed to enter into usage somewhere after 2031, and it is a long way until there.

File: tests/tests.cpp (hunk 3 above)
Comment (AssemblyJohn) [reply tests/tests.cpp:24]: Unintended.

**Comment by AssemblyJohn** [comment]
Update, tested locally, please re-review.

**Review by Pietfried — APPROVED** [review]

File: include/evse_security/evse_security.hpp [hunk 4]
```diff
@@ -142,6 +142,12 @@ class EvseSecurity {
     /// @param ocsp_response the actual OCSP data
     void update_ocsp_cache(const CertificateHashData& certificate_hash_data, const std::string& ocsp_response);
 
+    // TODO: Switch to path
```
Comment (hikinggrass) [inline include/evse_security/evse_security.hpp:145]: Looks like this already returns a optional path, is that what this comment meant?

Comment (AssemblyJohn) [reply include/evse_security/evse_security.hpp:145]: Opps, simply forgot a line.
